Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4661,6 +4668,8 @@ void WebPageProxy::continueNavigationInNewProcess(API::Navigation& navigation, W
             navigation = Ref { navigation },
             previousProvisionalLoadProcess = Ref { frame.provisionalLoadProcess() }
         ] () mutable {
+            if (!frame->isMainFrame())
+                frame->setHasPendingBackForwardItem();
```

Comment:
This value is later used to call setIsRootChildFrameItem, but there's no check to see if it is actually a root frame.  It could be a grandchild frame navigating to the domain of its parent, which would hit this code and the frame would not be becoming a root frame.